prov/util/src/util_shm2.c

```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright (c) 2016-2021 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
```

Comment:
I am not sure that shm and shm2 need different util. Can they share the same util? If not, can we move these to inside shm2?

Reply:
`smr2_calculate_size_offsets` will change b/c there response queue
`void smr2_cleanup(void)` will change b/c there won't be mutex locks
`struct smr2_region {` region structure definition will likely change 
`int smr2_create()` will change

The new util code is required AFAIK, but will look very different from SHM.

Reply:
Sounds like these functions is tied to how shm2 provider is implemented, so they should be inside shm2 provider. Right?

Reply:
We originally separated the shm region and map functions into the util code because we thought there would be other providers that might want to use them to implement an internal shm implementation directly. Since no one is using them and we are targeting the peer provider approach for shm offload, I think we can use this opportunity to move them under the shm provider directly and, once we merge the shm2 provider back with the shm provider, remove the util_shm.c code, replacing it with the one inside the provider itself.

Reply:
done